### PR TITLE
Disallow multi-line string literals

### LINKED ISSUE #19142: Fix 0xA and 0xD usages in Lexer
Current Implementation lexer rules don't adhere to spec. Because of this multiple line string line are allowed. 

```
string x = "foo
bar
"; // This should be a compilation error.
```

### PR DESCRIPTION
**Pull request by pubudu91**
## Purpose
Fixes #19142

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/resources/grammar/BallerinaLexer.g4 [hunk 1]
```diff
@@ -369,7 +369,7 @@ StringCharacters
 
 fragment
 StringCharacter
-    :   ~["\\]
+    :   ~(["\\] | '\u000A' | '\u000D')
```
[the remark this answers is not in the document]
Comment (pubudu91) [reply compiler/ballerina-lang/src/main/resources/grammar/BallerinaLexer.g4]: Yeah, that works. Will update the PR

File: tests/jballerina-unit-test/src/test/resources/test-src/types/string/string_negative.bal [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/string/string_negative.bal]: Just for clarity, shall we rename this file to something like `string_multi_line_negative.bal`?

**Comment by anupama-pathirage** [comment]
We need to update the standard library module.md pages and "how-to" pages in ballerina.io along with this change.

**Comment by pubudu91** [comment]
@anupama-pathirage Fixed a usage in the Module.md of jdbc module. Created https://github.com/ballerina-platform/ballerina-www/issues/1642 for checking the how-to pages.

**Pull request merged**